network/manager.go

```diff
@@ -475,39 +475,52 @@ func (nm *networkManager) DeleteEndpoint(networkID, endpointID string, epInfo *E
 		return err
 	}
 
-	err = nm.save()
-	if err != nil {
-		return err
-	}
-
 	return nil
 }
 
 func (nm *networkManager) DeleteEndpointState(networkID string, epInfo *EndpointInfo) error {
 	nw := &network{
-		Id:           networkID,
+		Id:           networkID, // CHECK: currently unused in stateless cni
+		HnsId:        epInfo.HNSNetworkID,
 		Mode:         opModeTransparentVlan,
 		SnatBridgeIP: "",
+		NetNs:        dummyGUID, // to trigger hns v2, windows
 		extIf: &externalInterface{
 			Name:       InfraInterfaceName,
 			MacAddress: nil,
 		},
 	}
 
 	ep := &endpoint{
-		Id:                       epInfo.Id,
+		Id:                       epInfo.EndpointID,
 		HnsId:                    epInfo.HNSEndpointID,
-		HostIfName:               epInfo.IfName,
+		HNSNetworkID:             epInfo.HNSNetworkID, // unused (we use nw.HnsId for deleting the network)
+		HostIfName:               epInfo.HostIfName,
 		LocalIP:                  "",
 		VlanID:                   0,
-		AllowInboundFromHostToNC: false,
+		AllowInboundFromHostToNC: false, // stateless currently does not support apipa
 		AllowInboundFromNCToHost: false,
 		EnableSnatOnHost:         false,
 		EnableMultitenancy:       false,
-		NetworkContainerID:       epInfo.Id,
+		NetworkContainerID:       epInfo.NetworkContainerID, // we don't use this as long as AllowInboundFromHostToNC and AllowInboundFromNCToHost are false
+		NetNs:                    dummyGUID,                 // to trigger hnsv2, windows
+		NICType:                  epInfo.NICType,
+		NICName:                  epInfo.IfName,
 	}
 	logger.Info("Deleting endpoint with", zap.String("Endpoint Info: ", epInfo.PrettyString()), zap.String("HNISID : ", ep.HnsId))
-	return nw.deleteEndpointImpl(netlink.NewNetlink(), platform.NewExecClient(logger), nil, nil, nil, nil, ep)
+	err := nw.deleteEndpointImpl(netlink.NewNetlink(), platform.NewExecClient(logger), nil, nil, nil, nil, ep)
+	if err != nil {
+		return err
+	}
+	if epInfo.NICType == cns.DelegatedVMNIC {
+		// we are currently assuming stateless is not running in linux
+		// CHECK: could this affect linux? (if it does, it could disconnect external interface, is that okay?)
+		// bad only when 1) stateless and 2) linux and 3) delegated vmnics exist
+		logger.Info("Deleting endpoint because delegated vmnic detected", zap.String("HNSNetworkID", nw.HnsId))
+		err = nm.deleteNetworkImpl(nw)
+		// no need to clean up state in stateless
```

Comment:
this is windows specfic and windows specific delete implementation should handle this.. lets move this code to windows specific file

Reply:
do you mean just having nm.deleteNetworkImpl in DeleteEndpointState (no if conditional)? I wanted to do this but we don't want to delete any network in the linux case and the deleteNetworkImpl on the linux side seems to have some side effects (can you explain what this function does)? I would modify deleteNetworkImpl for linux to be a noop but it still seems to be used in manager.go/restore and cnm delete network

Reply:
address later when we create an endpoint delete function to match EndpointCreate